functions/nodejs-apps.bash

```diff
@@ -202,3 +202,147 @@ nodered_setup() {
+  local introText="A MQTT-server is required for zigbee2mqtt. If you haven't installed one yet, please select <cancel> and come back after installing one (e.g. Mosquitto).\\n\\nZigbee2MQTT will be installed from the official repository.\\n\\nDuration is about 4 minutes..."
+  local installText="Zigbee2MQTT is installed from the official repository.\\n\\nPlease wait about 4 minutes..."
+  local uninstallText="Zigbee2MQTT is completely removed from the system."
+  local adapterText="Please select your zigbee adapter:"
+  local mqttUserText="\\nPlease enter your MQTT-User (default = openhabian):"
+  local mqttPWText="\\nIf your MQTT-server requires a password, please enter it here:"
+  local adapterArray=()
+  local my_adapters=""
+  local mqttDefaultUser="openhabian"
+  local mqttUser
+  local serverIP; serverIP="$(hostname -I)"; serverIP=${serverIP::-1} # remove trailing space
+  local installSuccessText="Setup was successful. Zigbee2MQTT is now up and running.\\n\\nFor further Zigbee-settings open frontend (in 2 minutes): \\nhttp://${serverIP}:8081.\n\nDocumentation of ZigBee2MQTT:\\nhttps://www.zigbee2mqtt.io/guide/configuration"
+  local updateSuccessText="Update successful. \\n\\nFor further Zigbee-settings open frontend (in 2 minutes): \\nhttp://${serverIP}:8081.\n\nDocumentation of Zigbee2MQTT:\\nhttps://www.zigbee2mqtt.io/guide/configuration"
+  local loopSel=1
+  local i=0
+
+  if [[ $1 == "remove" ]]; then
+    if [[ -n $INTERACTIVE ]]; then
+      if ! (whiptail --title "Zigbee2MQTT Uninstall" --yes-button "Continue" --no-button "Cancel" --yesno "$uninstallText" 7 80); then echo "CANCELED"; return 0; fi
+    fi
+    echo -n "$(timestamp) [openHABian] Removing Zigbee2MQTT service ... "
+    if ! cond_redirect systemctl stop zigbee2mqtt.service; then echo "FAILED (disable service)"; fi
+    if ! rm -f /etc/systemd/system/zigbee2mqtt.service; then echo "FAILED (remove service)"; fi
+    if cond_redirect systemctl -q daemon-reload; then echo "OK"; else  echo "FAILED (daemon-reload)"; return 1; fi
+
+    echo -n "$(timestamp) [openHABian] Uninstalling Zigbee2MQTT ... "
+    if ! cond_redirect npm uninstall zigbee2mqtt ; then echo "FAILED (npm uninstall)"; fi
+    if ! rm -rf /var/log/zigbee2mqtt; then echo "FAILED (remove log)"; fi
+    if rm -rf "/opt/zigbee2mqtt"; then echo "OK"; else echo "FAILED"; fi
+    
+    if [[ -n "$INTERACTIVE" ]]; then
+      whiptail --title "Zigbee2MQTT removed" --msgbox "Zigbee2MQTT was removed from your system." 7 80
+    fi
+    return 0;
+  fi
+  if [[ $1 != "install" ]]; then return 1; fi
+
+  # if a config file exists do only update and exit
+  if [[ -e "/opt/zigbee2mqtt/data/configuration.yaml" ]] ; then
+    if [[ -n $INTERACTIVE ]]; then
+      if ! (whiptail --title "Zigbee2MQTT installation" --yes-button "Continue" --no-button "Cancel" --yesno "$z2mInstalledText" 14 80); then echo "CANCELED"; return 0; fi
+    fi
+
+    echo -n "$(timestamp) [openHABian] Updating zigbee2mqtt... "
```

Comment:
Try to be consistent in your capitalization of "Zigbee2MQTT". Either all lowercase or all properly capitalized depending on which is most correct.

Reply:
done